test_base_fileurl_field/models/res_partner.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+
+    _inherit = 'res.partner'
+
+    name = fields.Char()
+    url_file = fields.FileURL(
+        storage_location='s3',
+        filename='url_file_fname',
+        storage_path='partner'
+    )
+    url_file_fname = fields.Char()
+
+    url_image = fields.FileURL(
+        storage_location='s3',
+        filename='url_image_fname',
+        storage_path='partner_image',
+    )
+    url_image_fname = fields.Char()
+
+    @api.constrains('url_file', 'url_file_fname')
```

Comment:
As I guess we would never want a duplicate url for a file, should/could this check be implemented on the field class definition ?

Reply:
It's unfortunately not possible at field level since the filename is implemented at model level. So the constraint must be implemented at model level as is the case there...

Reply:
Got it, thank you.